crates/router/src/services/api/request.rs

```diff
@@ -110,12 +168,15 @@ impl RequestBuilder {
     }
 
     pub fn header(mut self, header: &str, value: &str) -> Self {
-        self.headers.insert((header.into(), value.into()));
+        self.headers
+            .insert((header.into(), Box::new(value.to_string())));
         self
     }
 
-    pub fn headers(mut self, headers: Vec<(String, String)>) -> Self {
-        let mut h = headers.into_iter().map(|(h, v)| (h, v));
+    pub fn headers(mut self, headers: Vec<(String, Box<dyn HeaderValue>)>) -> Self {
```

Comment:
Why dynamic dispatch is required here, anyways this is a string or library's `HeaderValue`.  I don't see a `polymorphism` which needed to be solved in dynamic dispatch way.